bugbug/repository.py

```diff
@@ -452,9 +452,9 @@ def download_commits(repo_dir, date_from):
         directories = get_directories(commit.files)
 
         for directory in directories:
-            directories_touched_prev[commit.node] += directories_touched[directory]
+            directories_touched_prev[commit.node].update(directories_touched[directory])
```

Comment:
If the memory usage is too high after this change, we could still keep directories_touched_prev as defaultdict(int) making this code a little more complex.
If the memory usage is reasonable, we can avoid it.

Reply:
I think the memory usage would be high as we're storing a set of commit nodes for each commit in directories_touched_prev. And the size of these sets would increase as we traverse more commits.

And how can we check if the memory usage is high or not?

Reply:
Assuming you have less than 32 GB on your laptop, if the script still works after your changes without crashing because of running out of memory, then the memory usage is still reasonable.

Reply:
Basically, the memory usage is too high if you can't run the script on your laptopt.

Reply:
Okay.